cnxepub/tests/test_utils.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+import sys
+import unittest
+
+from lxml import etree
+
+
+IS_PY2 = sys.version_info.major == 2
+
+
+class TestSquashXMLToText(unittest.TestCase):
+
+    content = (
+        '<div data-type="description" xmlns="http://www.w3.org/1999/xhtml">'
+        'FOO '
+        '<p><math xmlns="http://www.w3.org/1998/Math/MathML"/></p>'
+        '<p><math xmlns="http://www.w3.org/1998/Math/MathML"/></p>'
+        ' BAR'
+        '</div>'
+    )
+
+    parser = etree.XMLParser(resolve_entities=True, encoding='ascii')
+
+    def setUp(self):
+        self.content = etree.fromstring(self.content, self.parser)
+
+    @property
+    def target(self):
+        from cnxepub.utils import squash_xml_to_text
+        return squash_xml_to_text
+
+    def test(self):
+        result = self.target(self.content, False)
+
+        expected = (
+            'FOO '
+            '<p xmlns="http://www.w3.org/1999/xhtml">'
+            '<math xmlns="http://www.w3.org/1998/Math/MathML"/></p>'
+            '<p xmlns="http://www.w3.org/1999/xhtml">'
+            '<math xmlns="http://www.w3.org/1998/Math/MathML"/></p>'
+            ' BAR'
+        )
+        assert result == expected
+
+    def test_with_namespace_removal(self):
+        result = self.target(self.content, True)
+
+        expected = 'FOO <p><math/></p><p><math/></p> BAR'
+        assert result == expected
+
+    def test_text_only(self):
+        txt = '<div>Hello Wórld!</div>'
+        expected = 'Hello Wórld!'
+        if IS_PY2:
```

Comment:
Next time you can just do `txt = u'<div>Hello Wórld!</div>'` so the meaning is the same in both python 2 and 3 instead of doing `.decode('utf-8')` if it's python 2.  Alternatively we can also do `from __future__ import unicode_literals`.